web/frontend/src/components/do-command/index.svelte

```diff
@@ -1,87 +1,102 @@
 <script lang="ts">
-
 import { commonApi } from '@viamrobotics/sdk';
 import { notify } from '@viamrobotics/prime';
 import { resourceNameToString } from '@/lib/resource';
 import { doCommand } from '@/api/do-command';
 import Collapse from '@/lib/components/collapse.svelte';
 import { useRobotClient } from '@/hooks/robot-client';
+import { getClientByType } from './get-client-by-type';
+import { Button, Label, SearchableSelect } from '@viamrobotics/prime-core';
+import { isUnimplementedError } from './errors';
 
 export let resources: commonApi.ResourceName.AsObject[];
 
 const { robotClient } = useRobotClient();
 
-let selectedComponent = '';
-let input = '';
+let selectedComponent = undefined as
+  | commonApi.ResourceName.AsObject
+  | undefined;
+let input = '{}';
 let output = '';
 let executing = false;
 
-const handleDoCommand = async (name: string, command: string) => {
-  if (!name || !command) {
+const handleDoCommand = async (type: string, name: string, command: string) => {
+  if (!type || !name || !command) {
```

Comment:
Should the above function args be typed as `string | undefined` or as optionals?

Reply:
Yeah that makes more sense, will clean that up.